pkg/cmd/server/api/types.go

```diff
@@ -252,23 +252,72 @@ type RequestHeaderIdentityProvider struct {
 	Headers []string
 }
 
-type OAuthRedirectingIdentityProvider struct {
+type GitHubIdentityProvider struct {
 	api.TypeMeta
 
 	// ClientID is the oauth client ID
 	ClientID string
 	// ClientSecret is the oauth client secret
 	ClientSecret string
-
-	// Provider contains the information about exactly which kind of oauth you're identifying with
-	Provider runtime.EmbeddedObject
 }
 
-type GoogleOAuthProvider struct {
+type GoogleIdentityProvider struct {
 	api.TypeMeta
+
+	// ClientID is the oauth client ID
+	ClientID string
+	// ClientSecret is the oauth client secret
+	ClientSecret string
 }
-type GitHubOAuthProvider struct {
+
+type OpenIDIdentityProvider struct {
 	api.TypeMeta
+
+	// CA is the optional trusted certificate authority bundle to use when making requests to the server
+	// If empty, the default system roots are used
+	CA string
+
+	// ClientID is the oauth client ID
+	ClientID string
+	// ClientSecret is the oauth client secret
+	ClientSecret string
+
+	// Scopes are the requested scopes sent with authorize requests.
+	// Defaults to "openid" if empty.
+	// Must contain "openid" if specified.
+	Scopes []string
+
+	// URLs to use to authenticate
+	URLs OpenIDURLs
+
+	// Claims mappings
+	Claims OpenIDClaims
+}
+
+type OpenIDURLs struct {
```

Comment:
yeah... I couldn't bring myself to name them "urlAuthorize", or to scatter the url fields throughout